projects/common/src/utilities/formatters/string/highlight.pipe.ts

```diff
@@ -1,20 +1,37 @@
 import { Pipe, PipeTransform } from '@angular/core';
+import { isArray } from 'lodash-es';
 import { assertUnreachable } from '../../lang/lang-utils';
 
 @Pipe({ name: 'htHighlight' })
 export class HighlightPipe implements PipeTransform {
   public transform(
     fullText: string,
-    highlightedText: string | undefined,
+    highlightSnippets?: TextHighlightConfig | TextHighlightConfig[],
```

Comment:
I think accepting a single text string here will make the most common use case easier (as seen from combo box change below). Not required though, just a shortcut.